In [4]:
# Data manipulation
import pandas as pd
import numpy as np

# Visualization
import matplotlib.pyplot as plt
import seaborn as sns



In [5]:
# Load Excel file

file_path = "../Data/Original.xlsx"

df = pd.read_excel(file_path)

# Display first rows
df.head()

/home/eduards/miniconda3/envs/torch/lib/python3.13/site-packages/openpyxl/worksheet/_reader.py:329: UserWarning: Conditional Formatting extension is not supported and will be removed
  warn(msg)


,Unnamed: 0,Unnamed: 1,Unnamed: 2,Unnamed: 3,Unnamed: 4,Unnamed: 5,05:00:00,Unnamed: 7,Unnamed: 8,Unnamed: 9,Unnamed: 10,Unnamed: 11,Unnamed: 12,Unnamed: 13,Unnamed: 14,Unnamed: 15,Unnamed: 16,Unnamed: 17,Unnamed: 18,Unnamed: 19,Unnamed: 20,Unnamed: 21,Unnamed: 22,Unnamed: 23,Unnamed: 24,Unnamed: 25,Unnamed: 26,Unnamed: 27,Unnamed: 28,Unnamed: 29,Unnamed: 30,Unnamed: 31,Unnamed: 32,Unnamed: 33,Unnamed: 34,Unnamed: 35,Unnamed: 36,Unnamed: 37,Unnamed: 38,Unnamed: 39,Unnamed: 40,Unnamed: 41,Unnamed: 42,Unnamed: 43,Unnamed: 44,Unnamed: 45,Unnamed: 46,Unnamed: 47,Unnamed: 48,Unnamed: 49,Unnamed: 50,Unnamed: 51,Unnamed: 52,Unnamed: 53,Unnamed: 54,Unnamed: 55,Unnamed: 56,Unnamed: 57,Unnamed: 58,Unnamed: 59,Unnamed: 60,Unnamed: 61,Unnamed: 62,Unnamed: 63,Unnamed: 64,Unnamed: 65,Unnamed: 66,Unnamed: 67,Unnamed: 68,Unnamed: 69,Unnamed: 70,Unnamed: 71,Unnamed: 72,Unnamed: 73,Unnamed: 74,Unnamed: 75,Unnamed: 76,Unnamed: 77,Unnamed: 78,Unnamed: 79,Unnamed: 80,Unnamed: 81,Unnamed: 82,Unnamed: 83,Unnamed: 84,Unnamed: 85,Unnamed: 86,Unnamed: 87,Unnamed: 88,Unnamed: 89,Unnamed: 90,Unnamed: 91,Unnamed: 92,Unnamed: 93,Unnamed: 94,Unnamed: 95,Unnamed: 96,Unnamed: 97,Unnamed: 98,NaN,Unnamed: 100,Unnamed: 101,Unnamed: 102,Unnamed: 103,Unnamed: 104,Unnamed: 105,Unnamed: 106,Unnamed: 107,Unnamed: 108,Unnamed: 109,Unnamed: 110,Unnamed: 111,Unnamed: 112,Unnamed: 113,Unnamed: 114,Unnamed: 115,Unnamed: 116,Unnamed: 117,Unnamed: 118,Unnamed: 119,Unnamed: 120,Unnamed: 121,Unnamed: 122,Unnamed: 123,Unnamed: 124,Unnamed: 125,Unnamed: 126,Unnamed: 127,Unnamed: 128,Unnamed: 129,Unnamed: 130,Unnamed: 131,Unnamed: 132,Unnamed: 133,Unnamed: 134,Unnamed: 135,Unnamed: 136,Unnamed: 137,Unnamed: 138,Unnamed: 139,Unnamed: 140,Unnamed: 141,Unnamed: 142,Unnamed: 143,Unnamed: 144,Unnamed: 145,Unnamed: 146,Unnamed: 147,Unnamed: 148,Unnamed: 149,Unnamed: 150,Unnamed: 151,Unnamed: 152,Unnamed: 153,Unnamed: 154,Unnamed: 155,Unnamed: 156,Unnamed: 157,Unnamed: 158,Unnamed: 159,Unnamed: 160,Unnamed: 161,Unnamed: 162,Unnamed: 163,Unnamed: 164,Unnamed: 165,Unnamed: 166,Unnamed: 167,Unnamed: 168,Unnamed: 169,Unnamed: 170,Unnamed: 171,Unnamed: 172,Unnamed: 173,Unnamed: 174,Unnamed: 175,Unnamed: 176,Unnamed: 177,|,Unnamed: 179,Unnamed: 180
0,Identificación del evento,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Lugar del evento,NaN,NaN,NaN,NaN,NaN,Coordenadas,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Coordenadas,NaN,Coordenadas,NaN,NaN,NaN,NaN,NaN,Explotador,NaN,NaN,NaN,NaN,Resumen,NaN,Datos de la operación,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Consecuencias,NaN,Lesiones,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Información del piloto,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Información del copiloto,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Datos de la aeronave,2,3,4,5,6,15,16,11,10,27,28,29,30,31,32,33,34,35,36,37,39,40,Estado de la aeronave,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Descripción de ocurrencias,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Conclusiones,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0,NaN,NaN,Análisis de factores,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Control interno,NaN,NaN,NaN,NaN,Control de finalización,NaN,NaN,Control de publicación,NaN,NaN
1,1,2,3,4,5,6,7,8,NaN,NaN,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,NaN,NaN,167,168,169,170,171,172,173,174,175,176,177
2,Código investigación,Clasificación,Matrícula,Fecha,Fecha UTC,Hora,Hora UTC,Condiciones de luz,Estado,Sigla estado,Departamento,Cod.D,Municipio,Cod.M,Localización frente a ad.,Aeródro

In [6]:
# Shape of dataset
print("Shape:", df.shape)

# Column names
print("\nColumns:")
print(df.columns)

# Data types
print("\nData Types:")
df.dtypes

Shape: (116, 181)

Columns:
Index([  'Unnamed: 0',   'Unnamed: 1',   'Unnamed: 2',   'Unnamed: 3',
         'Unnamed: 4',   'Unnamed: 5',       05:00:00,   'Unnamed: 7',
         'Unnamed: 8',   'Unnamed: 9',
       ...
       'Unnamed: 171', 'Unnamed: 172', 'Unnamed: 173', 'Unnamed: 174',
       'Unnamed: 175', 'Unnamed: 176', 'Unnamed: 177',            '|',
       'Unnamed: 179', 'Unnamed: 180'],
      dtype='object', length=181)

Data Types:


Unnamed: 0      object
Unnamed: 1      object
Unnamed: 2      object
Unnamed: 3      object
Unnamed: 4      object
                 ...  
Unnamed: 176    object
Unnamed: 177    object
|               object
Unnamed: 179    object
Unnamed: 180    object
Length: 181, dtype: object

- Vemos que la primera fila, no tiene un identificador para cada columna, esos identificador de columna son Unnamed. Esto no nos da mucha información así que deberíamos curar o modificar esto para tener una tabla de una forma mejor para trabajar. 
- Tenemos 116 filas y 181 columnas. 

In [8]:
# Missing values per column
missing_values = df.isnull().sum()

missing_values

Unnamed: 0        1
Unnamed: 1        2
Unnamed: 2        2
Unnamed: 3        2
Unnamed: 4        2
               ... 
Unnamed: 176     58
Unnamed: 177     65
|                73
Unnamed: 179     74
Unnamed: 180    105
Length: 181, dtype: int64

- Estos son los datos nulos por columna, vemos que algunas tienes varios datos nulos. Estos datos nulos no nos sirven para hacer cosas, entonces cuando limpieamos la tabla seguramente esos datos los eliminemos.
- ¿Esos datos nulos de que serán, hay una explicación de porqué están en nulos?

In [9]:
missing_percentage = (df.isnull().sum() / len(df)) * 100

missing_percentage.sort_values(ascending=False)

Unnamed: 169    99.137931
Unnamed: 168    99.137931
Unnamed: 31     98.275862
NaN             97.413793
Unnamed: 160    96.551724
                  ...    
Unnamed: 131     0.862069
Unnamed: 118     0.862069
Unnamed: 117     0.862069
Unnamed: 119     0.862069
Unnamed: 170     0.862069
Length: 181, dtype: float64

In [10]:
# Count duplicates
duplicates = df.duplicated().sum()

print("Duplicate rows:", duplicates)

Duplicate rows: 0
